projects/akita-ng-fire/src/lib/utils/sync-with-router.ts

```diff
@@ -0,0 +1,26 @@
+import { getPathParams } from './path-with-params';
+import { CollectionService, CollectionState } from '../collection/collection.service';
+import { distinctUntilChanged, filter, switchMap, shareReplay } from 'rxjs/operators';
+import { RouterQuery } from '@datorama/akita-ng-router-store';
+import { Observable } from 'rxjs';
+import { DocumentChangeAction } from '@angular/fire/firestore';
+
+export function syncWithRouter<Service extends CollectionService<CollectionState<E>>, E>(
+  this: Service,
+  routerQuery: RouterQuery
```

Comment:
You don't need to give the routerQuery as second parameter. You could do something like : 
```typescript
type ServiceWithRouter<E> = CollectionService<CollectionState<E>> & { routerQuery: RouterQuery };
export function syncWithRouter<Service extends ServiceWithRouter<E>, E>(
  this: Service
): Observable<DocumentChangeAction<E>[]> {}
```

The limitation doing that is that you force developer to use the naming `routerQuery`.